themes/default/content/blog/cloud-systems-part-three/index.md

```diff
@@ -0,0 +1,325 @@
+    policy_arn=aws.iam.ManagedPolicy.AMAZON_ECS_FULL_ACCESS)
+```
+
+We're going to be using [AWS Fargate](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/AWS_Fargate.html) to actually run our container. Fargate is a serverless compute platform designed for running containers without requiring you to provision and manage clusters of EC2 (Elastic Compute Cloud) instances. Since our website is containerized and very small, it's perfect for us. Services also require IAM roles, and in this case, we need to allow Fargate to execute our services and manage our tasks, so we create new roles for both and apply execution permissions to them.
+
+```python
+# Creating storage space to upload a docker image of our app to
+app_ecr_repo = aws.ecr.Repository("app-ecr-repo",
+    image_tag_mutability="MUTABLE")
+
+# Attaching an application life cycle policy to the storage
+app_lifecycle_policy = aws.ecr.LifecyclePolicy("app-lifecycle-policy",
+    repository=app_ecr_repo.name,
+    policy="""{
+        "rules": [
+            {
+                "rulePriority": 10,
+                "description": "Remove untagged images",
+                "selection": {
+                    "tagStatus": "untagged",
+                    "countType": "imageCountMoreThan",
+                    "countNumber": 1
+                },
+                "action": {
+                    "type": "expire"
+                }
+            }
+        ]
+    }""")
+```
+
+The last thing we need to do for our infrastructure before we can start deploying to it is create a repository on [Amazon ECR (Elastic Container Registry)](https://aws.amazon.com/ecr/) where our Docker image will live, then attach an application lifecycle policy to that repository. This makes sure that we expire and remove any untagged images in the repository, keeping things from geting clogged up.
+
+## Deploying a Dockerized Flask Application to ECS
+
+We're now ready to deploy our website and wire all of this up!
+
+```python
+flask_targetgroup = aws.lb.TargetGroup("flask-targetgroup",
+    port=80,
+    protocol="TCP",
+    target_type="ip",
+    vpc_id=app_vpc.id)
+
+flask_balancer = aws.lb.LoadBalancer("flask-balancer",
+    load_balancer_type="network",
+    internal=False,
+    security_groups=[],
+    subnets=[app_vpc_subnet.id])
+
+flask_listener = aws.lb.Listener("flask-listener",
+    load_balancer_arn=flask_balancer.arn,
+    port=80,
+    protocol="TCP",
+    default_actions=[aws.lb.ListenerDefaultActionArgs(
+        type="forward",
+        target_group_arn=flask_targetgroup.arn
+    )])
+```
+
+First, we need to make it possible for our Flask application to communicate with the internet. That requires three pieces of configuration: a target group for port 80, a load balancer to spread out incoming requests and make sure our website doesn't get overwhelmed as easily, and a listener to forward public traffic to the defined target group.
```

Comment:
I would add that the target group points to the VPC's endpoint and defines the port for that unknown-until-runtime, as people might be confused why there's more networking going on.